### ISSUE TITLE
Message lost if exceeding maxStringLength and emptyPropertyExclusionEnabled is true

### ISSUE THREAD

**Issue by waldeinburg** [vy/log4j2-logstash-layout#55]
If maxStringLength is exceeded and emptyPropertyExclusionEnabled is true, Jackson UTF8JsonGenerator will throw a JsonGenerationException: "Can not write a string, expecting field name (context: Object)".

**Comment by vy** [comment]
There is something pretty weird going on when `JsonGeneratorDelegate` is wrapped with a `TokenFilter` where `includeNull=false`. I could not reproduce your issue with the following snippet:

```java
@Test
public void test_maxStringLength_with_emptyPropertyExclusionEnabled() throws JsonProcessingException {

    // Create the event template.
    ObjectNode eventTemplateRootNode = JSON_NODE_FACTORY.objectNode();
    eventTemplateRootNode.put("message", "${json:message}");
    String eventTemplate = eventTemplateRootNode.toString();

    // Create the layout.
    int maxStringLength = 10;
    BuiltConfiguration configuration = ConfigurationBuilderFactory.newConfigurationBuilder().build();
    LogstashLayout layout = LogstashLayout
            .newBuilder()
            .setConfiguration(configuration)
            .setEventTemplate(eventTemplate)
            .setMaxStringLength(maxStringLength)
            .setEmptyPropertyExclusionEnabled(false)
            .build();

    // Create the log event.
    SimpleMessage message = new SimpleMessage(Strings.repeat('m', maxStringLength) + 'x');
    LogEvent logEvent = Log4jLogEvent
            .newBuilder()
            .setLoggerName(LogstashLayoutTest.class.getSimpleName())
            .setMessage(message)
            .build();

    // Check the serialized event.
    String serializedLogEvent = layout.toSerializable(logEvent);
    JsonNode rootNode = OBJECT_MAPPER.readTree(serializedLogEvent);
    assertThat(point(rootNode, "message").asText()).isEqualTo(message.getFormattedMessage());

}
```

Here `layout.toSerializable(logEvent)` doesn't thrown an exception, but the comparison still fails. I have filed FasterXML/jackson-core#609 for the issue.

**Comment by vy** [comment]
@waldeinburg, I have implemented a work around for FasterXML/jackson-core#609 in 634aec3. Would you mind checking if it works for you, please?

**Comment by waldeinburg** [comment]
The workaround works, thanks!

**Comment by vy** [comment]
I've just released v1.0.2 including this fix.
